### PR TITLE
Wallets balance

### PR DESCRIPTION
**Pull request by OlegGordiichuk**
- Add new field for shown of the wallet ETH balance.
- Update of the Wallet object.

@vikmeup could you take a look pls but do not merge pls. I would like to take a look in the morning.

Example of the work.

ezgif com-optimize-3

### CODE REVIEW

File: Trust/Wallet/Types/WalletObject.swift [hunk 1]
```diff
@@ -12,6 +12,7 @@ final class WalletObject: Object {
     @objc dynamic var createdAt: Date = Date()
     @objc dynamic var completedBackup: Bool = false
     @objc dynamic var mainWallet: Bool = false
+    @objc dynamic var value: String = ""
```
Comment (vikmeup) [inline Trust/Wallet/Types/WalletObject.swift]: name is `balance`

File: Trust/Wallet/Types/WalletInfo.swift [hunk 2]
```diff
@@ -94,3 +103,9 @@ extension WalletInfo: Equatable {
         return lhs.type.description == rhs.type.description
     }
 }
+
+extension WalletInfo {
+    static func format(value: String, server: RPCServer = .main) -> String {
```
Comment (vikmeup) [inline Trust/Wallet/Types/WalletInfo.swift]: can you remove default value?

File: Trust/Wallet/ViewControllers/WalletsViewController.swift [hunk 3]
```diff
@@ -13,35 +13,33 @@ class WalletsViewController: UITableViewController {
...
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-
-        fetch(animated: false)
+        fetch()
     }
 
-    func fetch(animated: Bool = true) {
+    func fetch() {
+        viewModel.fetchValues()
```
Comment (vikmeup) [inline Trust/Wallet/ViewControllers/WalletsViewController.swift]: call it `fetchBalances()`

File: Trust/Wallet/ViewModels/WalletsViewModel.swift [hunk 4]
```diff
@@ -4,13 +4,18 @@ import Foundation
 import UIKit
 import TrustCore
 
+protocol WalletsViewModelProtocol: class {
```
Comment (vikmeup) [inline Trust/Wallet/ViewModels/WalletsViewModel.swift]: does it have to be a protocol?

File: Trust/Wallet/ViewModels/WalletsViewModel.swift [hunk 5]
```diff
@@ -24,6 +29,31 @@ class WalletsViewModel {
...
+    func fetchValues() {
+
+        let operationQueue: OperationQueue = OperationQueue()
+        operationQueue.qualityOfService = .background
+
+        var valueProviders = [(CoinNetworkProvider, WalletObject)]()
+
+        for wallet in self.keystore.wallets {
+            guard let server = wallet.coin?.server else { continue }
+            valueProviders.append((CoinNetworkProvider(server: server, address: wallet.currentAccount.address, addressUpdate: EthereumAddress.zero), wallet.info))
```
Comment (vikmeup) [inline Trust/Wallet/ViewModels/WalletsViewModel.swift]: why address to update is `EthereumAddress.zero`?

File: Trust/Wallet/Views/WalletViewCell.swift [hunk 6]
```diff
@@ -12,10 +12,11 @@ final class WalletViewCell: UITableViewCell {
     @IBOutlet weak var infoButton: UIButton!
     @IBOutlet weak var glassesImageView: UIImageView!
     @IBOutlet weak var walletTypeImageView: UIImageView!
-    @IBOutlet weak var balanceLabel: UILabel!
-    @IBOutlet weak var addressLabel: UILabel!
+    @IBOutlet weak var title: UILabel!
+    @IBOutlet weak var address: UILabel!
```
Comment (vikmeup) [inline Trust/Wallet/Views/WalletViewCell.swift]: `address` => `subtitle`

File: Trust/Wallet/Views/WalletViewCell.swift [hunk 7]
```diff
@@ -12,10 +12,11 @@ final class WalletViewCell: UITableViewCell {
...
     @IBOutlet weak var identiconImageView: TokenImageView!
     @IBOutlet weak var selectedImageView: UIImageView!
+    @IBOutlet weak var value: UILabel!
```
Comment (vikmeup) [inline Trust/Wallet/Views/WalletViewCell.swift]: `value` => `balance`

File: Trust/Wallet/Views/WalletViewCell.swift [hunk 8]
```diff
@@ -24,12 +25,18 @@ final class WalletViewCell: UITableViewCell {
...
             }
-            addressLabel.text = model.title
-            balanceLabel.text = model.subbtitle
+            title.text = model.title
+            address.text = model.address
             glassesImageView.isHidden = !model.isWatch
             infoButton.tintColor = Colors.lightBlue
             identiconImageView.image = model.image
             selectedImageView.image = model.selectedImage
+            if !model.wallet.multiWallet {
```
Comment (vikmeup) [inline Trust/Wallet/Views/WalletViewCell.swift]: this has to be defined by the view model

`isBalanceHidden`,` balanceValue`

File: Trust/Wallet/Types/WalletObject.swift (hunk 1 above)
Comment (OlegGordiichuk) [reply Trust/Wallet/Types/WalletObject.swift]: done

File: Trust/Wallet/Types/WalletInfo.swift (hunk 2 above)
Comment (OlegGordiichuk) [reply Trust/Wallet/Types/WalletInfo.swift]: done

File: Trust/Wallet/ViewModels/WalletsViewModel.swift (hunk 4 above)
Comment (OlegGordiichuk) [reply Trust/Wallet/ViewModels/WalletsViewModel.swift]: yeah we should to use this as delegate to trigger tableview reload.

File: Trust/Wallet/Views/WalletViewCell.swift (hunk 8 above)
Comment (OlegGordiichuk) [reply Trust/Wallet/Views/WalletViewCell.swift]: done

File: Trust/Wallet/ViewControllers/WalletsViewController.swift (hunk 3 above)
Comment (OlegGordiichuk) [reply Trust/Wallet/ViewControllers/WalletsViewController.swift]: done

File: Trust/Wallet/Types/WalletInfo.swift [hunk 9]
```diff
@@ -34,6 +36,10 @@ struct WalletInfo {
         return accounts.count > 1
     }
 
+    var isBalanceHidden: Bool {
```
[the remark this answers is not in the document]
Comment (OlegGordiichuk) [reply Trust/Wallet/Types/WalletInfo.swift]: @vikmeup WalletInfoViewModel is used only in WalletInfoViewController. In our case it should be in WalletAccountViewModel that is used in WalletViewCell.

File: Trust/Wallet/Types/WalletInfo.swift [hunk 10]
```diff
@@ -76,6 +82,13 @@ struct WalletInfo {
         }
     }
 
+    var value: String {
```
[the remark this answers is not in the document]
Comment (OlegGordiichuk) [reply Trust/Wallet/Types/WalletInfo.swift]: done

**Comment by OlegGordiichuk** [comment]
One more test with different wallets:

simulator screen shot - iphone x - 2018-07-28 at 13 44 29

**Comment by vikmeup** [comment]
@madcake can you create an issue for android? to be similar/better, just to show balance

**Pull request merged**